neonvm/controllers/virtualmachine_qmp_queries.go

```diff
@@ -327,112 +631,26 @@ func QmpSyncMemoryToTarget(vm *vmv1.VirtualMachine, migration *vmv1.VirtualMachi
 			continue
 		}
 		// add memdev object
-		memdevId := strings.ReplaceAll(m.Data.Memdev, "/objects/", "")
-		cmd := []byte(fmt.Sprintf(`{"execute": "object-add", "arguments": {"id": "%s", "size": %d, "qom-type": "memory-backend-ram"}}`, memdevId, m.Data.Size))
-		_, err = target.Run(cmd)
+		memdevIdx, err := QmpMatchMemBackend(m.Data.Memdev)
 		if err != nil {
 			return err
 		}
-		// now add pc-dimm device
-		cmd = []byte(fmt.Sprintf(`{"execute": "device_add", "arguments": {"id": "%s", "driver": "pc-dimm", "memdev": "%s"}}`, m.Data.Id, memdevId))
-		_, err = target.Run(cmd)
+		err = QmpAddMemoryBackend(target, memdevIdx, m.Data.Size)
 		if err != nil {
-			// device_add command failed... so try remove object that we just created
-			cmd = []byte(fmt.Sprintf(`{"execute": "object-del", "arguments": {"id": "%s"}}`, m.Data.Memdev))
-			target.Run(cmd)
 			return err
 		}
-	}
-
-	return nil
-}
-
-func QmpPlugMemoryToRunner(ip string, port int32, size int64) error {
-	memoryDevices, err := QmpQueryMemoryDevices(ip, port)
-	if err != nil {
-		return err
-	}
-	plugged := int32(len(memoryDevices))
-
-	mon, err := QmpConnect(ip, port)
-	if err != nil {
-		return err
-	}
-	defer mon.Disconnect()
-
-	// add memdev object for next slot
-	// firstly check if such object already present to avoid repeats
-	cmd := []byte(fmt.Sprintf(`{"execute": "qom-list", "arguments": {"path": "/objects/memslot%d"}}`, plugged+1))
-	_, err = mon.Run(cmd)
-	if err != nil {
-		// that mean such object wasn't found, let's add it
-		cmd = []byte(fmt.Sprintf(`{"execute": "object-add", "arguments": {"id": "memslot%d", "size": %d, "qom-type": "memory-backend-ram"}}`, plugged+1, size))
-		_, err = mon.Run(cmd)
+		// now add pc-dimm device
+		err = QmpAddMemoryDevice(target, memdevIdx)
 		if err != nil {
+			// device_add command failed... so try remove object that we just created
+			_ = QmpDelMemoryBackend(target, memdevIdx)
```

Comment:
What if the delete operation fails here? No error handling

Reply:
Which commit you mean? I think I'm looking at the latest version.

Reply:
You are looking at the latest version, but at the isolated 2nd commit. It's gone in the third commit https://github.com/neondatabase/autoscaling/pull/704/commits/98856c0f0287c899a8d9d831bd73020778a95086#diff-acc89b7771770ec55e48e065d33e10243a15901645298c0b241e06f7f67ed369L392

GitHub interface is quite confusing to do per-commit review :disappointed: